packages/simcore-sdk/tests/unit/test_node_ports_v2_port_schemas.py

```diff
@@ -0,0 +1,157 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+# pylint:disable=no-member
+# pylint:disable=protected-access
+# pylint:disable=too-many-arguments
+
+
+import json
+from typing import Any, Dict
+from unittest.mock import AsyncMock
+
+import jsonschema
+import pytest
+from pydantic.error_wrappers import ValidationError
+from simcore_sdk.node_ports_v2.port import Port
+from simcore_sdk.node_ports_v2.utils_schemas import (
+    jsonschema_validate_data,
+    jsonschema_validate_schema,
+)
+
+# HELPERS --------------------------------------------------------------------------------------
+
+
+def get_model_config_example(model_cls, label) -> Dict[str, Any]:
+    return next(
+        v for v in model_cls.Config.schema_extra["examples"] if v["label"] == label
+    )
+
+
+# TESTS --------------------------------------------------------------------------------------
+def test_json_validation_with_array():
+    # with array
+    schema = {
+        "title": "list[number]",
+        "type": "array",
+        "items": {"type": "number"},
+    }
+
+    data = [1, 2, 3]
+    jsonschema.validate(data, schema)
+
+    jsonschema_validate_schema(schema)
+    jsonschema_validate_data(data, schema)
+
+
+def test_json_validation_with_object():
+
+    # with object
+    schema = {
+        "title": "an object named A",
+        "type": "object",
+        "properties": {
+            "i": {"title": "Int", "type": "integer", "default": 3},
+            "b": {"title": "Bool", "type": "boolean"},
+            "s": {"title": "Str", "type": "string"},
+        },
+        "required": ["b", "s"],
+    }
+
+    data = {"b": True}
+    with pytest.raises(jsonschema.ValidationError):
+        jsonschema.validate(data, schema)
+
+    jsonschema_validate_schema(schema)
+    with pytest.raises(jsonschema.ValidationError):
+        jsonschema_validate_data(data, schema)
+
+    data = {"b": True, "s": "foo"}
+    jsonschema.validate(data, schema)
+
+    assert data == {"b": True, "s": "foo"}
+    assert jsonschema_validate_data(data, schema, return_with_default=True) == {
+        "b": True,
+        "s": "foo",
+        "i": 3,
+    }
+    assert jsonschema_validate_data(data, schema, return_with_default=False) == {
+        "b": True,
+        "s": "foo",
+    }
+
+
+async def test_port_with_array(mocker):
+    mocker.patch.object(Port, "_node_ports", new=AsyncMock())
+
+    # arrays
+    port_info = {
+        "label": "array_numbers",
+        "description": "Some array of numbers",
+        "type": "ref_contentSchema",
+        "contentSchema": {
+            "title": "list[number]",
+            "type": "array",
+            "items": {"type": "number"},
```

Comment:
question here: how do yo handle List[SimcoreFileLink]?
Or is that currently not allowed?
same question with the arbitrary object?

Reply:
mmmhh very good point!

Reply:
- with an arbitrary object it would not be a problem. 
- with ``List[SimcoreFileLink]`` would be treated as data itself. This is a special case though in which we want to identify a specific structure and act on in. It is important to make a distinction between a data payload that is just passed to the service and the one that requires some pre-processing (like ``SimcoreFileLink`` and the rest of family of links) before passing it to the service (e.g. downloading and converting into a path etc).

``List[BaseFileLink]`` is indeed very important and real use case but right now it is out of the scope of this PR (among other things because there is no front-end implementation of it). Here we focus on data payload that does not require any action (as described above).

Reply:
ok as discussed in a pair review, it would at least make sense to forbid usage of BaseFileLink for now so that it does not happen